.github/workflows/main.yml

```diff
@@ -9,75 +9,63 @@ on:
     branches:
       - '**'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  build_and_test:
+  build_core:
+    name: Build Stencil
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
```

Comment:
a small idea I had: each of the jobs that we define here in `main.yml` starts with the same `actions/checkout` bit, and then within each of their separate yml files (e.g. `build-core/action.yml` for this one) has to start with the `get-core-dependencies` action.

Would it be helpful for us to move the `checkout` action into the `get-core-dependencies` action, possibly renaming it to something like `checkout-and-install` (or some better name than that) ? I was just thinking that if we did that then we'd avoid having to put this SHA in four different places, and could just put it in one place instead.

Reply:
> Would it be helpful for us to move the checkout action into the get-core-dependencies action

I had the same idea, the problem is that putting that in a separate composite action/job will run the checkout on another VM, so we won't have the code, actions, etc. we need to run a job. GitHub actually errors here asking if we checked out the code or not when we move this 😢 

But let me double check that in a "throwaway PR" - I was moving fast and breaking things during JEDI days, let me double check my understanding and I'll get back to you on this

Reply:
I double checked this last night and confirmed it's a bit of a "chicken and egg" problem. It'd be nice to abstract the checkout action into the `get-core-dependencies` job, but to get that job, we'd need to checkout the repo first 🙃 For now I think we'll need to leave this as is (the good news is that at least we should be able to ctrl-f/replace instances of the SHA when it comes time to update this)

Reply:
sounds straightforward enough!